src/Ddeboer/DataImport/Workflow.php

```diff
@@ -269,118 +104,45 @@ public function process()
-    protected function filterItem($item, \SplPriorityQueue $filters)
+    public function setSkipItemOnFailure($skipItemOnFailure)
     {
-        // SplPriorityQueue must be cloned because it is a stack and thus drops
-        // elements each time it is iterated over.
-        foreach (clone $filters as $filter) {
-            if (false == $filter->filter($item)) {
-                return false;
-            }
-        }
+        $this->skipItemOnFailure = $skipItemOnFailure;
 
-        // Return true if no filters failed
-        return true;
+        return $this;
     }
 
-    /**
-     * Convert the item
-     *
-     * @param string $item Original item values
-     *
-     * @return array                   Converted item values
-     * @throws UnexpectedTypeException
-     */
-    protected function convertItem($item)
+    public function getName()
     {
-        foreach ($this->itemConverters as $converter) {
-            $item = $converter->convert($item);
-            if ($item && !(is_array($item) || ($item instanceof \ArrayAccess && $item instanceof \Traversable))) {
-                throw new UnexpectedTypeException($item, 'false or array');
-            }
-
-            if (!$item) {
-                return $item;
-            }
-        }
-
-        if ($item && !(is_array($item) || ($item instanceof \ArrayAccess && $item instanceof \Traversable))) {
-            throw new UnexpectedTypeException($item, 'false or array');
-        }
-
-        foreach ($this->valueConverters as $property => $converters) {
-            // isset() returns false when value is null, so we need
-            // array_key_exists() too. Combine both to have best performance,
-            // as isset() is much faster.
-            if (isset($item[$property]) || array_key_exists($property, $item)) {
-                foreach ($converters as $converter) {
-                    $item[$property] = $converter->convert($item[$property]);
-                }
-            }
-        }
-
-        return $item;
+        return $this->name;
     }
 
-    /**
-     * Get item converter that takes care of mapping
-     *
-     * @return MappingItemConverter
-     */
-    protected function getMappingItemConverter()
+    private function getOrderedSteps()
     {
-        // Find mapping item converter
-        $converters = \array_filter(
-            $this->itemConverters,
-            function ($converter) {
-                return $converter instanceof MappingItemConverter;
-            }
-        );
+        $tmp = $this->steps;
+        $steps = [];
 
-        if (count($converters) > 0) {
-            // Return first mapping item converter that we encounter
-            reset($converters);
-            $converter = current($converters);
-        } else {
-            // Create default converter
-            $converter = new MappingItemConverter();
-            $this->addItemConverter($converter);
-        }
-
-        return $converter;
-    }
+        ksort($tmp);
```

Comment:
So: would it cause too much memory usage if you clone the SplPriorityQueue before iterating over it?